src/helpers/userBehaviour.js

```diff
@@ -1,26 +1,51 @@
 const HAS_VISITED_FEEDBACK_PAGE = 'HAS_VISITED_FEEDBACK_PAGE';
 const NAVIGATION_COUNT = 'NAVIGATION_COUNT';
+const SHOW_CHART_PATTERNS = 'SHOW_CHART_PATTERNS';
 
-const safelySetLocalStorageItem = (name, value) => {
+const setItem = (name, item) => {
   try {
-    window.localStorage.setItem(name, value);
-  } catch (error) {
-    console.warn(`Could not set local storage item ${name} with value ${value}: ${error}`);
+    const serializedItem = JSON.stringify(item);
+
+    window.localStorage.setItem(name, serializedItem);
+  } catch (err) {
+    console.warn(err);
+  }
+};
+
+const getItem = (name) => {
+  try {
+    const serializedItem = window.localStorage.getItem(name);
+
+    if (serializedItem === null) return undefined;
+
+    return JSON.parse(serializedItem);
+  } catch (err) {
+    console.warn(err);
+
+    return undefined;
```

Comment:
Updated the logic here to wrap setItem/getItem to simplify the specific functions below.